src/Php74/composer.json

```diff
@@ -0,0 +1,36 @@
+{
+    "name": "symfony/polyfill-php74",
+    "type": "library",
+    "description": "Symfony polyfill backporting some PHP 7.4+ features to lower PHP versions",
+    "keywords": ["polyfill", "shim", "compatibility", "portable"],
+    "homepage": "https://symfony.com",
+    "license": "MIT",
+    "authors": [
+        {
+            "name": "Ion Bazan",
+            "email": "ion.bazan@gmail.com"
+        },
+        {
+            "name": "Nicolas Grekas",
+            "email": "p@tchwork.com"
+        },
+        {
+            "name": "Symfony Community",
+            "homepage": "https://symfony.com/contributors"
+        }
+    ],
+    "require": {
+        "php": ">=5.3.3",
+        "symfony/polyfill-mbstring": "^1.3"
```

Comment:
should be removed, that's an optional dep, we don't want to force the polyfill to everyone

Reply:
@nicolas-grekas I disagree. 
Calling `mb_str_split()` while using PHP 7.3 with `symfony/polyfill-php74` installed and without `ext-mbstring` will cause an error. Until https://github.com/composer/composer/issues/7557 is resolved, there is no better solution than requiring `symfony/polyfill-mbstring` for this package and using `replaces` key in project root when `ext-mbstring` is installed.

Reply:
The dep is optional, because when you have ext-mbstring, the polyfill is not needed.
The responsibility is on the consumer. If they have to require the polyfill, they can. Or they can require only ext-mbstring.
That's always been our policy.

Reply:
OK, I'll remove it then. I just seen it in https://github.com/symfony/polyfill/blob/6afbb6c3cfa6ca217c3e07f8aa7306c56b546792/src/Intl/Idn/composer.json#L20 and followed that.

Reply:
Looks like we missed this during the review in #149, sorry for the confusion it creates now.

Reply:
what we could do is having `symfony/polyfill-php74` polyfilling `mb_str_split` only if `mb_substr` is available. This way, you get the polyfill for it only if the other parts of mbstring are available (either through the mbstring polyfill or through the native extension).

And `symfony/polyfill-mbstring` should polyfill it all the time

Reply:
@nicolas-grekas, thank you for explanation :smiley: Removed the polyfill requirement.
@stof Good idea, implemented :+1: